packages/site/cms-site/src/blocks/DamFileDownloadLinkBlock.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+
+import { DamFileDownloadLinkBlockData } from "../blocks.generated";
+import { withPreview } from "../iframebridge/withPreview";
+import { PropsWithData } from "./PropsWithData";
+
+interface Props extends PropsWithData<DamFileDownloadLinkBlockData> {
+    children: React.ReactNode;
+    title?: string;
+}
+
+export const DamFileDownloadLinkBlock = withPreview(
+    ({ data: { file, openFileType }, children, title }: Props) => {
+        if (file === undefined) {
+            return <>{children}</>;
+        }
+
+        if (openFileType === "DOWNLOAD") {
+            return <a href={`${process.env.API_URL}/dam/files/download/${file.id}/${file.name}`}>{children}</a>;
```

Comment:
I think instead of building the URL in the site, you should add a `createFileDownloadUrl()` method to the FilesService and add the downloadUrl to the block (loading it in the API part of the block).